diff -Nurp nvidia-settings.orig/src/Makefile nvidia-settings/src/Makefile
--- nvidia-settings.orig/src/Makefile	2012-05-02 17:06:02.000000000 +0200
+++ nvidia-settings/src/Makefile	2012-05-17 09:09:32.341638054 +0200
@@ -51,7 +51,7 @@ ifndef GTK_CFLAGS
 endif
 
 ifndef GTK_LDFLAGS
-  GTK_LDFLAGS         := $(shell $(PKG_CONFIG) --libs gtk+-2.0)
+  GTK_LDFLAGS         := $(shell $(PKG_CONFIG) --libs gtk+-2.0) -ldl
 endif
 
 
