Index: nvidia-settings-331.20/src/Makefile
===================================================================
--- nvidia-settings-331.20.orig/src/Makefile	2013-11-06 11:29:44.000000000 -0500
+++ nvidia-settings-331.20/src/Makefile	2013-12-09 11:16:50.070936284 -0500
@@ -49,7 +49,7 @@
 endif
 
 ifndef GTK_LDFLAGS
-  GTK_LDFLAGS         := $(shell $(PKG_CONFIG) --libs gtk+-2.0)
+  GTK_LDFLAGS         := $(shell $(PKG_CONFIG) --libs gtk+-2.0) -ldl
 endif
 
 ifndef VDPAU_CFLAGS
