Author: Andreas Beckmann <debian@abeckmann.de>
Description: build a shared library for libXNVCtrl
Bug-Debian: https://bugs.debian.org/666909

Index: nvidia-settings-410.57/src/Makefile
===================================================================
--- nvidia-settings-410.57.orig/src/Makefile
+++ nvidia-settings-410.57/src/Makefile
@@ -113,6 +113,9 @@ endif
 XNVCTRL_DIR             ?= libXNVCtrl
 XNVCTRL_MAKEFILE        ?= Makefile
 XNVCTRL_ARCHIVE         ?= $(XNVCTRL_DIR)/libXNVCtrl.a
+XNVCTRL_SHARED          ?= $(XNVCTRL_DIR)/libXNVCtrl.so.0
+#XNVCTRL_LIB            ?= $(XNVCTRL_ARCHIVE)
+XNVCTRL_LIB             ?= $(XNVCTRL_SHARED)
 XCONFIG_PARSER_DIR      ?= XF86Config-parser
 COMMON_UTILS_DIR        ?= common-utils
 COMMON_UNIX_DIR         ?= common-unix
@@ -282,9 +285,9 @@ NVIDIA_SETTINGS_install: $(NVIDIA_SETTIN
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
 
 $(eval $(call DEBUG_INFO_RULES, $(NVIDIA_SETTINGS)))
-$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(XNVCTRL_ARCHIVE)
+$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(XNVCTRL_LIB)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
-	    -rdynamic -o $@ $(OBJS) $(XNVCTRL_ARCHIVE) $(LIBS)
+	    -rdynamic -o $@ $(OBJS) $(XNVCTRL_LIB) $(LIBS)
 
 $(eval $(call DEBUG_INFO_RULES, $(GTK2LIB)))
 $(GTK2LIB).unstripped: $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
@@ -292,7 +295,7 @@ $(GTK2LIB).unstripped: $(GTK2_OBJS) $(XC
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK2LIB_SONAME) \
 	    $(GTK2_OBJS) $(IMAGE_OBJS) $(XCP_OBJS) \
-	    $(XNVCTRL_ARCHIVE) $(LIBS) $(GTK2_LIBS)
+	    $(XNVCTRL_LIB) $(LIBS) $(GTK2_LIBS)
 
 ifdef BUILD_GTK3LIB
 $(eval $(call DEBUG_INFO_RULES, $(GTK3LIB)))
@@ -301,7 +304,7 @@ $(GTK3LIB).unstripped: $(GTK3_OBJS) $(XC
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK3LIB_SONAME) \
 	    $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) \
-	    $(XNVCTRL_ARCHIVE) $(LIBS) $(GTK3_LIBS)
+	    $(XNVCTRL_LIB) $(LIBS) $(GTK3_LIBS)
 endif
 
 # define the rule to build each object file
@@ -309,7 +312,7 @@ $(foreach src,$(SRC),$(eval $(call DEFIN
 $(foreach src,$(XCP_SRC),$(eval $(call DEFINE_OBJECT_RULE,TARGET,$(src))))
 
 # define the rule to build $(XNVCTRL_ARCHIVE)
-$(XNVCTRL_ARCHIVE): build-xnvctrl
+$(XNVCTRL_ARCHIVE) $(XNVCTRL_SHARED): build-xnvctrl
 
 build-xnvctrl:
 	@$(MAKE) -C $(XNVCTRL_DIR) -f $(XNVCTRL_MAKEFILE)
Index: nvidia-settings-410.57/src/libXNVCtrl/Makefile
===================================================================
--- nvidia-settings-410.57.orig/src/libXNVCtrl/Makefile
+++ nvidia-settings-410.57/src/libXNVCtrl/Makefile
@@ -57,13 +57,21 @@ OBJS = $(call BUILD_OBJECT_LIST,$(SRC))
 .PHONY: clean
 
 all: $(LIBXNVCTRL)
+all: libXNVCtrl.so
 
 $(LIBXNVCTRL) : $(OBJS)
 	$(AR) ru $@ $(OBJS)
 
+libXNVCtrl.so: $(OBJS)
+	$(RM) $@ $@.*
+	$(CC) -shared -Wl,-soname=$@.0 -o $@.0.0.0 $(LDFLAGS) $^ -lXext -lX11
+	ln -s $@.0.0.0 $@.0
+	ln -s $@.0 $@
+
 # define the rule to build each object file
 $(foreach src,$(SRC),$(eval $(call DEFINE_OBJECT_RULE,TARGET,$(src))))
 
 clean:
 	rm -rf $(LIBXNVCTRL) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d
+	rm -f libXNVCtrl.so libXNVCtrl.so.*
