Author: Andreas Beckmann <debian@abeckmann.de>
Description: let clean recurse into libXNVCtrl

Index: nvidia-settings-410.57/src/Makefile
===================================================================
--- nvidia-settings-410.57.orig/src/Makefile
+++ nvidia-settings-410.57/src/Makefile
@@ -318,6 +318,7 @@ build-xnvctrl:
 $(foreach src,$(XCP_SRC),$(eval $(call DEFINE_OBJECT_RULE,TARGET,$(src))))
 
 clean clobber:
+	$(MAKE) -C $(XNVCTRL_DIR) clean
 	rm -rf $(NVIDIA_SETTINGS) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
 		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
