Author: Andreas Beckmann <debian@abeckmann.de>
Description: let clean recurse into libXNVCtrl

--- a/src/Makefile
+++ b/src/Makefile
@@ -299,6 +299,7 @@ $(foreach src,$(SRC),$(eval $(call DEFIN
 $(foreach src,$(XCP_SRC),$(eval $(call DEFINE_OBJECT_RULE,TARGET,$(src))))
 
 clean clobber:
+	$(MAKE) -C $(XNVCTRL_DIR) clean
 	rm -rf $(NVIDIA_SETTINGS) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
 		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
